spynnaker/pyNN/models/spike_source/spike_source_poisson_vertex.py

```diff
@@ -541,71 +521,14 @@ def kiss_seed(self, vertex_slice):
-    def read_parameters_from_machine(
-            self, transceiver, placement, vertex_slice):
-
-        # locate SDRAM address where parameters are stored
-        poisson_params = placement.vertex.poisson_param_region_address(
-            placement, transceiver)
-        seed_array = transceiver.read_memory(
-            placement.x, placement.y, poisson_params + SEED_OFFSET_BYTES,
-            SEED_SIZE_BYTES)
-        self.__kiss_seed[vertex_slice] = struct.unpack_from("<4I", seed_array)
-
-        # locate SDRAM address where the rates are stored
-        poisson_rate_region_sdram_address = (
-            placement.vertex.poisson_rate_region_address(
-                placement, transceiver))
-
-        # get size of poisson params
-        size_of_region = get_rates_bytes(vertex_slice, self.__data["rates"])
-
-        # get data from the machine
-        byte_array = transceiver.read_memory(
-            placement.x, placement.y,
-            poisson_rate_region_sdram_address, size_of_region)
-
-        # For each atom, read the number of rates and the rate parameters
-        offset = 0
-        for i in range(vertex_slice.lo_atom, vertex_slice.hi_atom + 1):
-            n_values = struct.unpack_from("<I", byte_array, offset)[0]
-            offset += 4
-
-            # Skip reading the index, as it will be recalculated on data write
-            offset += 4
-
-            (_start, _end, _next, is_fast_source, exp_minus_lambda,
-             sqrt_lambda, isi, time_to_next_spike) = _PoissonStruct.read_data(
-                 byte_array, offset, n_values)
-            offset += _PoissonStruct.get_size_in_whole_words(n_values) * 4
-
-            # Work out the spikes per tick depending on if the source is
-            # slow (isi), fast (exp) or faster (sqrt)
-            is_fast_source = is_fast_source == 1.0
-            spikes_per_tick = numpy.zeros(len(is_fast_source), dtype="float")
-            spikes_per_tick[is_fast_source] = numpy.log(
-                exp_minus_lambda[is_fast_source]) * -1.0
-            is_faster_source = sqrt_lambda > 0
-            # pylint: disable=assignment-from-no-return
-            spikes_per_tick[is_faster_source] = numpy.square(
-                sqrt_lambda[is_faster_source])
-            slow_elements = isi > 0
-            spikes_per_tick[slow_elements] = 1.0 / isi[slow_elements]
-
-            # Convert spikes per tick to rates
-            machine_time_step = (
-                globals_variables.get_simulator().machine_time_step)
-            self.__data["rates"].set_value_by_id(
-                i,
-                spikes_per_tick *
-                (MICRO_TO_SECOND_CONVERSION / float(machine_time_step)))
-
-            # Store the updated time until next spike so that it can be
-            # rewritten when the parameters are loaded
-            self.__data["time_to_spike"].set_value_by_id(
-                i, time_to_next_spike)
+    def update_kiss_seed(self, vertex_slice, seed):
```

Comment:
Any reason that this information needs to be stored at the app vertex level rather than at the machine vertex level?  The machine vertex is by definition "by vertex slice" so storing there instead should work...

Reply:
highly possible, but out of scope of THIS pr. there's possibly lots of stuff that could move from app to machine vertex once all the stuff has been moved. but to avoid feature creep. it was not done in this pr

Reply:
OK, fair enough